### PR TITLE
feat: allow floating menu tooltip to be triggered onhover

### PR DESCRIPTION
**Pull request by ZacharyStair**
Closes carbon-design-system/carbon-addons-ics#

I like the floating tooltip in that they can 'break out' of their containers (which may have 'overflow hidden' for instance), which was an issue i was running into using the basic TooltipHover components.

This adds a boolean flag that can be used to turn the Tooltip component into a floating 'tooltiphover'-esque component

I considered making a higher-order component to track mouse state but it would ultimately need basically everything in this pr as well, so the HOC wasn't saving any code

#### Changelog

**New**

- openOnHover prop for Tooltip component

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -45,9 +47,15 @@ export default class Tooltip extends Component {
     }
   };
 
-  handleFocus = direction => {
+  /**
+   * Function to toggle open state, calling getTriggerPosition when opening
+   * @param {boolean} open - what state to toggle to
+   * @memberof Tooltip
+   */
```
Comment (stpCollabr8nLstn) [inline src/components/Tooltip/Tooltip.js]: 🥇 nice!

File: package.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
     "storybook": "start-storybook -s ./public -p 9000",
     "build-storybook": "build-storybook -s ./public",
     "start": "npm run build:icons && npm run storybook",
-    "test": "jest -u --coverage"
+    "test": "jest --coverage"
```
Comment (ZacharyStair) [inline package.json]: the `-u` flag is only meant to be used when you're sure that the snapshots should be updated, if you run it every time you'll never know when the snapshots failed

File: src/components/Tooltip/Tooltip-test.js [hunk 3]
```diff
@@ -61,24 +61,33 @@ describe('Tooltip', () => {
 
   describe('events', () => {
     it('changes state on click', () => {
-      const wrapper = mount(<Tooltip triggerText="Tooltip" />);
+      const wrapper = mount(<Tooltip showIcon={false} triggerText="Tooltip" />);
       const trigger = wrapper.find('.bx--tooltip__trigger');
-      trigger.simulate('click');
-      expect(wrapper.state().open).toEqual(false);
```
Comment (ZacharyStair) [inline src/components/Tooltip/Tooltip-test.js]: some of these tests didn't look like they were working right before, i fixed up the logic so they're actually testing what they say they're testing

**Comment by ZacharyStair** [comment]
@stpCollabr8nLstn added some unit testing, how does it look?

File: package.json (hunk 2 above)
Comment (stpCollabr8nLstn) [reply package.json]: 👍

**Pull request merged**